### PR TITLE
errorModal: refactor to use React Component

### PR DESCRIPTION
**Pull request by mareklibra**
No API or functional change.

The component is intended to be reused by other commits.

### CODE REVIEW

File: frontend/public/components/modals/error-modal.jsx [hunk 1]
```diff
@@ -2,14 +2,13 @@ import * as React from 'react';
...
-        <ModalTitle>Error</ModalTitle>
-        <ModalBody>{error}</ModalBody>
-        <ModalFooter inProgress={false} errorMessage=""><button type="button" onClick={(e) => cancel(e)} className="btn btn-default">OK</button></ModalFooter>
-      </div>
-    );
-  }
-);
+export const ModalErrorContent = ({ error, title = 'Error', cancel }) => (
+  <div className="modal-content">
+    <ModalTitle>{title}}</ModalTitle>
```
Comment (spadgett) [inline frontend/public/components/modals/error-modal.jsx]: Extra `}`

```suggestion
    <ModalTitle>{title}</ModalTitle>
```

Comment (mareklibra) [reply frontend/public/components/modals/error-modal.jsx]: done

File: frontend/public/components/modals/error-modal.jsx [hunk 2]
```diff
@@ -2,14 +2,13 @@ import * as React from 'react';
...
-      </div>
-    );
-  }
-);
+export const ModalErrorContent = ({ error, title = 'Error', cancel }) => (
+  <div className="modal-content">
+    <ModalTitle>{title}</ModalTitle>
+    <ModalBody>{error}</ModalBody>
+    <ModalFooter inProgress={false} errorMessage="">
+      <button type="button" onClick={(e) => cancel(e)} className="btn btn-default">OK</button>
```
Comment (kyoto) [inline frontend/public/components/modals/error-modal.jsx]: Just `onClick={cancel}`?

File: frontend/public/components/modals/error-modal.jsx [hunk 3]
```diff
@@ -2,14 +2,13 @@ import * as React from 'react';
...
+    </ModalFooter>
+  </div>);
+
+export const errorModal = createModalLauncher(props => <ModalErrorContent {...props} />);
```
Comment (kyoto) [inline frontend/public/components/modals/error-modal.jsx]: Could this be just `createModalLauncher(ModalErrorContent)`?

Comment (mareklibra) [reply frontend/public/components/modals/error-modal.jsx]: done

File: frontend/public/components/modals/error-modal.jsx (hunk 2 above)
Comment (mareklibra) [reply frontend/public/components/modals/error-modal.jsx]: done

**Comment by rawagner** [comment]
@spadgett @kyoto if it looks good to you, approve and lgtm please :)

**Pull request merged**